components/script/dom/htmlimageelement.rs

```diff
@@ -603,15 +603,38 @@ impl HTMLImageElement {
                 }
             }
         }
+        let mut max_den = 0f64; //initial value of current_den in Step 5
+        let mut max_index = 0; //initial value of result_index in Step 5
+        let mut current_index = 0;
         let img_sources = &mut vec![];
         for outer_index in 0..len {
             if !repeat_indices.contains(&outer_index) {
                 img_sources.push(&source_set.image_sources[outer_index]);
+                // set max density and index
+                let den = img_sources[current_index].descriptor.den.unwrap();
+                if max_den < den {
+                    max_den = den;
+                    max_index = current_index;
+                }
+                current_index = current_index + 1;
```

Comment:
We can write this more idiomatically ask:
```rust
let mut max = (0f64, 0);
for image_source in &source_set.image_sources {
    if repeat_indices.contains(&index) {
        continue;
    }
    let den = image_source.descriptor.den.unwrap();
    if max.0 < den {
        max = (den, img_sources.len());
    }
    img_sources.push(image_source);
}
```

Reply:
Do we have to generate `index` here too by enumerating like the next suggestion ?

Reply:
Oh, right. Yes, I missed that one.